eks/sync.go

```diff
@@ -250,37 +255,47 @@ func upgradeCoreDNS(
 ) error {
 	logger := logging.GetProjectLogger()
 
-	targetImage := fmt.Sprintf("602401143452.dkr.ecr.%s.amazonaws.com/eks/coredns:v%s", awsRegion, coreDNSVersion)
-	currentImage, err := getCurrentDeployedCoreDNSImage(clientset)
-	if err != nil {
-		return err
-	}
-	if currentImage == targetImage {
-		logger.Info("Current deployed version matches expected version. Skipping coredns update.")
-		return nil
-	}
-
 	logger.Info("Confirming compatibility of coredns configuration with latest version.")
-	corednsConfigMap, err := getCorednsConfigMap(clientset)
-	if err != nil {
-		return err
-	}
 	// Need to check config for backwards incompatibility if updating to version >= 1.7.0. The keyword upstream was
 	// removed in 1.7 series of coredns, but is used in earlier versions.
-	compareVal, err := semverStringCompare(coreDNSVersion, "1.7.0-eksbuild.1")
+	compareVal170, err := semverStringCompare(coreDNSVersion, "1.7.0-eksbuild.1")
 	if err != nil {
 		return err
 	}
 	// Looking for 1 or 0 here, since we want to know when coreDNSVersion is >= 1.7.0
```

Comment:
Since annotated versions are considered `<`, the comparison is actually against `1.7.0-eksbuild.1`, which is `< 1.7.0`. I don't know if it really matters in this comment, but it strikes me as slightly inaccurate. Do I understand correctly?

Reply:
I think it could be argued that it doesn't matter for the sake of this comment, since `1.7.0-eksbuild.1` is still `1.7.0` colloquially.

Reply:
This doesn't really matter, because all the EKS coredns versions are always published with the annotation, so the comment is accurate in the context of EKS.